### PR TITLE
[stable/airflow] provide additional labels configuration

### PR DESCRIPTION
**Pull request by corymurphy**
#### What this PR does / why we need it:
This change will allow you to optionally specify a label for the web Ingress resource. This is a useful feature for ingress controllers such as traefik that allow you to filter Ingresses based on arbitrary labels.

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/airflow/Chart.yaml
Comment (thesuperzapper) [inline stable/airflow/Chart.yaml]: Any time we add an API (e.g. a helm Value), we do a minor version bump, for example, in this case `7.12.0`

Meaning you will also need to update: [UPGRADE.md](https://github.com/helm/charts/blob/master/stable/airflow/UPGRADE.md)

File: stable/airflow/README.md
Comment (thesuperzapper) [inline stable/airflow/README.md]: I believe this value should be: `ingress.web.labels` to be more consistent with the other values in the chart.

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
Thanks for the PR!

Other than my comments could you also implement this feature for `flower`? (That is, create `ingress.flower.labels`)

While you're there, you might also want to implement values for `flower` which do what `ingress.web.precedingPaths` and `ingress.web.succeedingPaths` do for `webserver`. (So we can have feature parity between `ingress-flower` and `ingress-webserver`)

File: stable/airflow/Chart.yaml
Comment (corymurphy) [reply stable/airflow/Chart.yaml]: I incremented the minor version and updated `UPGRADE.md`, thanks!

File: stable/airflow/README.md
Comment (corymurphy) [reply stable/airflow/README.md]: I have updated the value to be more consistent, thanks for the feedback! I kept the wording in the documentation the same, just to indicate that there are default labels already defined. I can change this if you don't think it's appropriate.

**Comment by corymurphy** [comment]
> 
> 
> Thanks for the PR!
> 
> Other than my comments could you also implement this feature for `flower`? (That is, create `ingress.flower.labels`)
> 
> While you're there, you might also want to implement values for `flower` which do what `ingress.web.precedingPaths` and `ingress.web.succeedingPaths` do for `webserver`. (So we can have feature parity between `ingress-flower` and `ingress-webserver`)

Thank you for the quick feedback! I have implemented the `labels` feature for `ingress-flower` and also added `ingress.flower.precedingPaths` and `ingress.flower.succeedingPaths`. I wasn't sure if I should document `ingress.flower.precedingPaths` and `ingress.flower.succeedingPaths` in the `README.md`, or if I should leave it as `<see values.yaml>`.

File: stable/airflow/README.md
Comment (thesuperzapper) [inline stable/airflow/README.md]: These are not needed, as we only put the top level values in the README.md

In this case, `ingress.web.labels` would be included in `ingress.web.*`.

File: stable/airflow/UPGRADE.md
Comment (thesuperzapper) [inline stable/airflow/UPGRADE.md:3]: Please add a link inside the README.md to this heading.

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
See my comments 

I will merge this once #23882 has merged, (you will need to rebase once that happens)

**Comment by corymurphy** [comment]
> 
> 
> [APPROVALNOTIFIER] This PR is **NOT APPROVED**
> 
> This pull-request has been approved by: _[corymurphy](https://github.com/helm/charts/pull/23867#)_
> To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **dwardu89**, **liyinan926**, **rk295**, **thesuperzapper** after the PR has been reviewed.
> You can assign the PR to them by writing `/assign @dwardu89 @liyinan926 @rk295 @thesuperzapper` in a comment when ready.
> 
> The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=helm%2Fcharts).
> Needs approval from an approver in each of these files:
> 
>     * **[incubator/rundeck/OWNERS](https://github.com/helm/charts/blob/master/incubator/rundeck/OWNERS)**
> 
>     * **[incubator/sparkoperator/OWNERS](https://github.com/helm/charts/blob/master/incubator/sparkoperator/OWNERS)**
> 
>     * **[stable/airflow/OWNERS](https://github.com/helm/charts/blob/master/stable/airflow/OWNERS)**
> 
>     * **[stable/gangway/OWNERS](https://github.com/helm/charts/blob/master/stable/gangway/OWNERS)**
> 
> 
> Approvers can indicate their approval by writing `/approve` in a comment
> Approvers can cancel approval by writing `/approve cancel` in a comment

This was an unintentional backmerge, working to clean the history.

**Comment by thesuperzapper** [comment]
@corymurphy can you please squash your commits?

**Comment by thesuperzapper** [comment]
@corymurphy two things

1. you need to sign-off your commits, [see here](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work)
1. please keep the commit message title short, for example: `[stable/airflow] provide additional labels configuration`

**Comment by corymurphy** [comment]
> 
> 
> @corymurphy two things
> 
>     1. you need to sign-off your commits, [see here](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work)
> 
>     2. please keep the commit message title short, for example: `[stable/airflow] provide additional labels configuration`

My apologies, didn't realize the commit had the wrong email. Thanks you for your patience!

**Pull request merged**
